plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/TextRecordWriter.java

```diff
@@ -50,6 +51,13 @@ public TextRecordWriter(Path path, JobConf jobConf, Properties properties, boole
         rowSeparator = rowSeparatorByte;
         output = path.getFileSystem(jobConf).create(path, Reporter.NULL);
         compressedOutput = createCompressedStream(jobConf, output, isCompressed);
+
+        Optional<String> skipHeaderLine = Optional.ofNullable(properties.getProperty("skip.header.line.count"));
+        if (skipHeaderLine.isPresent()) {
```

Comment:
Please verify that when `skipHeaderLine.isEmpty()` then `textHeaderWriter.isEmpty() too.

Reply:
👍

Reply:
Ok I will throw the following exception if `textHEaderWriter` is not empty:
`TrinoException(HIVE_WRITER_DATA_ERROR, "TextHeaderWriter should be empty when skip.header.line.count is not set");`

Reply:
Do not throw `TrinoException(HIVE_WRITER_DATA_ERROR` here. 
Just use `checkArgument(...)`. It is an assertion to catch programmer errors.

Reply:
Ok I went with a little different approach, hopefully it is fine

Reply:
I think we should pass `textHeaderWriter` here if-and-only-if `skip.header.line.count==1`. So you should extend condition in `RecordFileWriter` and taighten the verification here.

Reply:
I didn't extend the condition because I thought it is not a good place to check `skip.header.line.count` parameter, it would look messy. I think it should only be checked in `TextRecordWriter` where it belongs.  For the same reason I was not able to add throwing exception in the `TextRecordWriter` when `skipHeaderLine` is not present and `textHeaderWriter` is present - it could be present even if this parameter is not set.

Reply:
👍